astropy/coordinates/tests/test_sky_coord.py

```diff
@@ -1809,3 +1809,22 @@ class MultipleAliasesFrame(BaseCoordinateFrame):
     assert isinstance(coord.transform_to('alias_2').frame, MultipleAliasesFrame)
 
     ftrans.unregister(frame_transform_graph)
+
+
+@pytest.mark.parametrize("kwargs, error_message", [
+    (
+        {"ra": 1, "dec": 1, "distance": 1 * u.pc, "unit": "deg"},
+        "Unit 'deg' (angle) could not be applied to 'distance'",
+    ),
+    (
+        {"rho": 1 * u.m, "phi": 1, "z": 1 * u.m, "unit": "deg", "representation_type": "cylindrical"},
+        "Unit 'deg' (angle) could not be applied to 'rho'",
+    ),
+])
+def test_passing_inconsistent_coordinates_and_units_raises_helpful_error(kwargs, error_message):
+    # https://github.com/astropy/astropy/issues/10725
+    with pytest.raises(ValueError) as excinfo:
+        SkyCoord(**kwargs)
+    assert (
```

Comment:
It is nitpicking but avoid the blacken parenthesis on its own line - the general goal is to stick to the style of a given file. This through-out.

Though here you can avoid the whole assert by doing `with pytest.raises(ValueError, match=error_message):`

Reply:
This is not because of black, it's a leftover because `error_message` used to be a long string, and I forgot to change it.

Will use the `match=` parameter instead though, much nicer!